src/Codeception/Command/SelfUpdate.php

```diff
@@ -118,189 +106,15 @@ public function execute(InputInterface $input, OutputInterface $output)
     }
 
     /**
-     * Checks whether the provided version is current.
-     *
-     * @param string $version The version number to check.
-     * @param string $latestVersion Latest stable version
-     * @return boolean Returns True if a new version is available.
-     */
-    private function isOutOfDate($version, $latestVersion)
-    {
-        return -1 != version_compare($version, $latestVersion, '>=');
-    }
-
-    /**
-     * @return string
-     */
-    private function getLatestStableVersion()
-    {
-        $stableVersions = $this->filterStableVersions(
-            $this->getGithubTags(self::GITHUB_REPO)
-        );
-
-        return array_reduce(
-            $stableVersions,
-            function ($a, $b) {
-                return version_compare($a, $b, '>') ? $a : $b;
-            }
-        );
-    }
-
-    /**
-     * @param array $tags
-     * @return array
-     */
-    private function filterStableVersions($tags)
-    {
-        return array_filter($tags, function ($tag) {
-            return preg_match('/^[0-9]+\.[0-9]+\.[0-9]+$/', $tag);
-        });
-    }
-
-    /**
-     * Returns an array of tags from a github repo.
-     *
-     * @param  string $repo The repository name to check upon.
-     * @return array
-     */
-    protected function getGithubTags($repo)
-    {
-        $jsonTags = $this->retrieveContentFromUrl(
-            'https://api.github.com/repos/' . $repo . '/tags'
-        );
-
-        return array_column(json_decode($jsonTags, true), 'name');
-    }
-
-    /**
-     * Retrieves the body-content from the provided URL.
+     * Returns base url of phar file for current PHP version
      *
-     * @param  string $url
      * @return string
-     * @throws \Exception if status code is above 300
      */
-    private function retrieveContentFromUrl($url)
+    protected function getPharUrl()
     {
-        $ctx = $this->prepareContext($url);
-
-        $body = file_get_contents($url, 0, $ctx);
-
-        if (isset($http_response_header)) {
-            $code = substr($http_response_header[0], 9, 3);
-            if (floor($code / 100) > 3) {
-                throw new \Exception($http_response_header[0]);
-            }
-        } else {
-            throw new \Exception('Request failed.');
+        return self::PHAR_URL;
+        if (version_compare(PHP_VERSION, '7.2.0', '<')) {
```

Comment:
Unreachable code

Reply:
Oops, this is a bug. Thanks for noticing it.

Reply:
Fixed